### ISSUE TITLE
Snabb config set errors for large values of max fragments per packet

### ISSUE THREAD

**Issue by takikawa** [Igalia/snabbswitch#697]
If you pass `snabb config set` the `/softwire-config/internal-interface/reassembly/max-fragments-per-packet` path with a large (unrealistic) value, it can error in the following ways:

```
$ sudo ./snabb config set -s snabb-softwire-v1 7612 "/softwire-config/internal-interface/reassembly/max-fragments-per-packet" "332157"
hugetlb mmap failed (Cannot allocate memory), falling back.
lib/ctable.lua:142: mmap failed: Cannot allocate memory
stack traceback:
        core/main.lua:149: in function <core/main.lua:147>
        [C]: in function 'error'
        lib/ctable.lua:142: in function 'calloc'
        lib/ctable.lua:157: in function 'resize'
        lib/ctable.lua:119: in function 'new'
        apps/lwaftr/ctable_wrapper.lua:41: in function 'initialize_frag_table'
        apps/lwaftr/ipv6_apps.lua:47: in function 'new'
        core/app.lua:305: in function <core/app.lua:304>
        core/app.lua:332: in function 'apply_config_actions'
        apps/config/follower.lua:42: in function 'handle_actions_from_leader'
        apps/config/follower.lua:55: in function 'method'
        ...
        program/lwaftr/setup.lua:506: in function 'run_worker'
        [string "require('program.lwaftr.setup').run_worker({[..."]:1: in function 'f'
        core/main.lua:58: in function <core/main.lua:43>
        [C]: in function 'xpcall'
        core/main.lua:211: in main chunk
        [C]: at 0x004536b0
        [C]: in function 'pcall'
        core/startup.lua:3: in main chunk
        [C]: in function 'require'
        [string "require "core.startup""]:1: in main chunk
```

```
sudo ./snabb config set -s snabb-softwire-v1 9754 "/softwire-config/internal-interface/reassembly/max-fragments-per-packet" "3321572178"
apps/lwaftr/fragmentv6_hardened.lua:244: size of C type is unknown or too large at line 2
stack traceback:
        core/main.lua:149: in function <core/main.lua:147>
        [C]: in function 'typeof'
        apps/lwaftr/fragmentv6_hardened.lua:244: in function 'initialize_frag_table'
        apps/lwaftr/ipv6_apps.lua:47: in function 'new'
        core/app.lua:305: in function <core/app.lua:304>
        core/app.lua:332: in function 'apply_config_actions'
        apps/config/follower.lua:42: in function 'handle_actions_from_leader'
        apps/config/follower.lua:55: in function 'method'
        core/app.lua:89: in function 'with_restart'
        core/app.lua:454: in function 'thunk'
        core/histogram.lua:98: in function 'breathe'
        core/app.lua:410: in function 'main'
        program/lwaftr/setup.lua:506: in function 'run_worker'
        [string "require('program.lwaftr.setup').run_worker({[..."]:1: in function 'f'
        core/main.lua:58: in function <core/main.lua:43>
        [C]: in function 'xpcall'
        core/main.lua:211: in main chunk
        [C]: at 0x004536b0  
        [C]: in function 'pcall'
        core/startup.lua:3: in main chunk
        [C]: in function 'require'
        [string "require "core.startup""]:1: in main chunk
```

Maybe the protocol should specify a smaller integer type for this field? (this came up in random testing)

Note: there are other paths that trigger similar allocation errors with large values, e.g., `/softwire-config/internal-interface/reassembly/max-packets`

**Comment by kbara** [comment]
Yes, I think those are important edge cases; thank you for finding them.

**Comment by dpino** [comment]
@kbara can you take care of this? It seems a bug in the fragmentation/reassembly code.

**Comment by kbara** [comment]
I'd consider it a bug in specifying validation ranges at the yang level, rather than in the fragmentation/reassembly code, though it could be seen either way. If we fix it at the fragmentation/reassembly level, we need to introduce mechanisms to explicitly say when setting a value is out of range, etc - which is doable, but is a fair bit more baroque than just specifying ranges in our model.

**Comment by dpino** [comment]
@kbara sounds reasonable. Still, are you interested in fixing it even if it's a Yang level? It doesn't seem hard.

**Comment by tsyesika** [comment]
After speaking to @kbara and taking a better look at this issue it seems the main problem is that we're not checking integer ranges and we definitely could be doing. I think the best way to do this would be to have the leader perform checks either from explicit ranges if they're given or the implicit ranges the type has. This would allow us to ensure out of range values aren't set and a better error message is produced for the implicit type overflows.

I'm unsure if we'll need a new version of the yang model, we already an updated version with the `br-address` changes, I don't see why all these changes causing incompatibilities couldn't be rolled into the same update.

any thoughts @wingo?

**Comment by wingo** [comment]
Patch in https://github.com/Igalia/snabb/commits/range-validation; will wait until #861 goes in until PR'ing it though.

**Comment by wingo** [comment]
Fixed by #862.
